### PR TITLE
Cleanup SystemIndexMigration tests

### LINKED ISSUE #84245: Clean up Feature Migration tests
There's some cleanup to be done in `FeatureMigrationIT` and related classes.

- [x] Convert `static` `preMigrationHook` and `postMigrationHook` to use instance members, retrieved from the `internalCluster()`
- [x] Use Hamcrest matchers as described [in this comment](https://github.com/elastic/elasticsearch/pull/84192/files#r812224067) for better failure output
- [x] Rename `createSimpleSettings` to `createSettings`
- [x] Rename `createSimpleMapping` to `createMapping`
- [x] add support for unknown fields https://github.com/elastic/elasticsearch/pull/84258#pullrequestreview-890890422

### PR DESCRIPTION
**Pull request by pgomulka**
A follow up after #84192
- refactor the static state in TestPlugin to be an instance
- refactor assertions to use hamcrest
- remove Simple from methods as it is not meaningful
- refactor xcontent tests to support unknown fields

closes #84245

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Review by AthenaEryma — APPROVED** [review]
LGTM, thanks for taking care of this!

**Review by rjernst — COMMENTED** [review]
I left a suggestion and a question

File: server/src/test/java/org/elasticsearch/upgrades/SystemIndexMigrationTaskParamsXContentTests.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ protected SystemIndexMigrationTaskParams doParseInstance(XContentParser parser)
 
     @Override
     protected boolean supportsUnknownFields() {
-        return false;
+        return true;
```
[the remark this answers is not in the document]
Comment (gwbrown) [reply server/src/test/java/org/elasticsearch/upgrades/SystemIndexMigrationTaskParamsXContentTests.java:31]: This was recommended on the backport; see [this review](https://github.com/elastic/elasticsearch/pull/84258#pullrequestreview-890890422).

File: modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java [hunk 2]
```diff
@@ -115,23 +117,30 @@
...
-    public void setupTestPlugin() {
-        TestPlugin.preMigrationHook.set((state) -> Collections.emptyMap());
-        TestPlugin.postMigrationHook.set((state, metadata) -> {});
+    public void setupTestPlugin() throws InterruptedException {
+        TestPlugin testPlugin = getTestPlugin();
+        testPlugin.preMigrationHook.set((state) -> Collections.emptyMap());
+        testPlugin.postMigrationHook.set((state, metadata) -> {});
+    }
+
+    public TestPlugin getTestPlugin() {
```
[the remark this answers is not in the document]
Comment (pgomulka) [reply modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/AbstractFeatureMigrationIntegTest.java:126]: good spot, I did not notice that. 
fixed

**Pull request merged**